File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        curSum = maxSum =nums[0]
        
        for num in nums[1:]:
            curSum = max(num, curSum+num)
            maxSum = max(curSum, maxSum)
        
        return maxSum
        
        
#         maxSum = -math.inf
        
#         for i in range(len(nums)):
#             currentSum = 0
#             for j in range(i,len(nums)):
#                 currentSum+=nums[j]
#                 maxSum = max(currentSum, maxSum)
        
#         return maxSum
        

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curSubArray = nums[0];
        int maxSubArray = nums[0];
        for(int i=1; i < nums.length; i++){
            int num = nums[i];
            curSubArray = Math.max(num, curSubArray+num);
            maxSubArray = Math.max(curSubArray, maxSubArray);
        }
        return maxSubArray;
        
    }
}